kiosk/kiosk_browser/main_widget.py

```diff
@@ -69,6 +72,23 @@ def closeEvent(self, event):
 
     # Private
 
+    def _toggle_virtual_keyboard(self, physical_keyboard_is_available):
+        if physical_keyboard_is_available:
+            if self._keyboardWidget:
+                logging.info("Physical keyboard available, disabling virtual keyboard")
+                # TODO: make it somehow visible in the GUI that the vkb is
+                # disabled? Will make it less confusing if someone forgets to
+                # unplug a keyboard and expects the vkb to come up when using the RC.
+                self._keyboardWidget.deleteLater()
+                self._keyboardWidget = None
+
+        else:
+            if self._keyboardWidget is None:
+                logging.info("No physical keyboard, enabling virtual keyboard")
+                self._keyboardWidget = KeyboardWidget(self)
+            else:
+                logging.warning("Physical keyboard just detected, but KeyboardWidget already initialized - this should not happen.")
```

Comment:
Why not? I use the system with just a remote and the keyboard widget gets initialized, then I plug in a regular physical keyboard. This seems like a plausible path?

Maybe I misunderstand something.